Automattic/wp-calypso
# Jetpack Cloud: prompt for server credentials on the fix/ignore single threat

#### Changes proposed in this Pull Request

* Ask for server credentials on the fix single threat and fix all threat dialogs (if we don't have them).

#### Implementation notes

I created a component called `ServerCredentialsWizardDialog` that before displaying its content, will verify if we have the server credentials of the user. If there are no server credentials, the user will have to enter them before proceeding to the next step. Basically, I abstracted what we were already doing in the "Fix all threats" dialog so we can use it in the "Fix/Ignore threat" dialog.

#### Questions

* What should be the title of the dialog on the fix/ignore single threat case? The fix case looks fine to me, but the ignore case doesn't. See screenshots at the bottom.
* The server credentials form has a copy that fits well in the "Fix all threats" flow, but doesn't look great on the other flows. How should we adjust that?

#### Testing instructions

For this to work, you will have to delete your site's `main` credentials from wordpress.com (`https://wordpress.com/settings/security/<yourSiteHere>`). After this, you can proceed with the rest of the instructions.

* Go to `http://jetpack.cloud.localhost:3000/`
* Select a site that has Scan and a couple of threats
* Click on `[Fix all]` button
* Verify the server credentials form looks well (copy, title, colors, elements)
* Enter your credentials
* Verify the "Fix all threats" step of the process looks fine
* Close the dialog (don't fix the threats)
* Click on any `[Fix threat]` button to fix a single threat
* Verify the server credentials form looks well (copy, title, colors, elements)
* Enter your credentials
* Verify the "Fix threat" step of the process looks fine
* Close the dialog (don't fix the threat)
* Click on any `[Ignore threat]` button to ignore a single threat
* Verify you don't see the server credentials form
* Verify the "Ignore threat" step of the process looks fine

1151678672052943-as-1175580852050409

#### Demo

**Fix all threat**
![image](https://user-images.githubusercontent.com/3418513/82093618-dd19ac80-96d1-11ea-849a-f9c3d9f49a91.png)

**Fix threat**
![image](https://user-images.githubusercontent.com/3418513/82093646-e99e0500-96d1-11ea-9c52-4a9f45205982.png)

![IgnoreDoesNotRequireCredentials](https://user-images.githubusercontent.com/3418513/81966094-5c827f80-95ef-11ea-9914-69b06607ce22.gif)
